src/metabase/actions/actions.clj

```diff
@@ -484,3 +484,24 @@
   {:database (or database (when table-id (:id (cached-database-via-table-id table-id))))
    :table-id table-id
    :row      (update-keys (or row row-arg) u/qualified-name)})
+
+;;;; `:table.row/create-or-update` -- similar to common but with additional :key field
```

Comment:
Since typically `key` will refer to the key in a map entry, this is a poor name.

How about `row-key`? Small difference, but avoids muscle memory kicking in when reading code which uses it.

It also avoids shadowing a Clojure built-in when we destructure.

Reply:
how do you feel about `row-id`?